src/main/kotlin/uk/gov/justice/digital/hmpps/visitscheduler/service/PrisonerService.kt

```diff
@@ -85,16 +80,20 @@ class PrisonerService(
 
   fun getPrisoner(prisonerId: String): PrisonerDto? {
     val prisonerSearchResultDto = prisonerOffenderSearchClient.getPrisoner(prisonerId)
-    val incentiveLevelCode = prisonerSearchResultDto?.currentIncentive?.level?.code
-    var incentiveLevel: IncentiveLevel? = null
-    incentiveLevelCode?.let {
-      incentiveLevel = IncentiveLevel.getIncentiveLevel(it)
 
-      if (incentiveLevel == null) {
-        LOG.error("Incentive level - $it for prisoner - $prisonerId not available in IncentiveLevel enum.")
+    return prisonerSearchResultDto?.let {
+      val incentiveLevelCode = prisonerSearchResultDto.currentIncentive?.level?.code
+      var incentiveLevel: IncentiveLevel? = null
+      incentiveLevelCode?.let {
+        incentiveLevel = IncentiveLevel.getIncentiveLevel(it)
+
+        if (incentiveLevel == null) {
+          LOG.error("Incentive level - $it for prisoner - $prisonerId not available in IncentiveLevel enum.")
+        }
       }
+
+      PrisonerDto(prisonerId, prisonerSearchResultDto.category, incentiveLevel, prisonCode = prisonerSearchResultDto.prisonId)
     }
```

Comment:
bit confusing git hubs compare I think most of this code is the same..

Reply:
Updated this method to return a null if a record is not found